brush-parser/src/ast.rs

```diff
@@ -73,6 +76,30 @@ impl Display for AndOrList {
     }
 }
 
+/// Iterator for [`AndOrList`].
+pub struct AndOrListIter {
+    list: VecDeque<AndOr>,
+}
+
+impl Iterator for AndOrListIter {
+    type Item = AndOr;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.list.pop_front()
+    }
+}
+
+impl IntoIterator for AndOrList {
+    type Item = AndOr;
+    type IntoIter = AndOrListIter;
+
+    fn into_iter(self) -> Self::IntoIter {
```

Comment:
It's hard to know what the requested use case is--hence my question on the issue--but my thought is that most AST observers are probably immutably reading it vs. consuming it. We could certain use it ourselves internally.

What do you think about starting by adding `fn iter(&self) -> ...` on `AndOrList` instead of `IntoIterator` as the first step? If we wanted to keep it simple perhaps we could have it return `impl Iterator<Item = &AndOr>` and use something like `std::iter::once`, `.chain()`, etc. to stitch together the first element with the remaining ones? I haven't tried it myself so I'm not sure if there's issues with that approach...

Reply:
I don't like the idea of making a method that shares the same name with one from a notable trait. Seems cheeky and deceptive to a caller.

Reply:
Fair point. No deception desired.

I think what I'm looking for is a way to implement `IntoIterator` on `&AndOrList` to allow read-only iteration. Can you see a way to do that?

Reply:
Yes, we can do that I think.

Reply:
Actually, I don't think we can do non-consuming iterators unless they are cloned first, which makes sense, because pulling from an iterator gives the borrowed item, which makes it consuming by nature.